packages/components/nodes/memory/AgentMemory/mysqlSaver.ts

```diff
@@ -54,80 +50,64 @@ CREATE TABLE IF NOT EXISTS ${this.tableName} (
     }
 
     async getTuple(config: RunnableConfig): Promise<CheckpointTuple | undefined> {
-        await this.setup()
+        const dataSource = await this.getDataSource()
```

Comment:
thanks for the PR!

Wondering why are we initializing a new data source for every action? this will most likely cause memory leak

Reply:
ignore me, actually with current implementation every run will causes constructor to be initialized, hence its the same